sugar_test.go

```diff
@@ -44,6 +44,13 @@ func TestSugarWith(t *testing.T) {
 		{[]interface{}{}, []zapcore.Field{}, nil},
 		{[]interface{}{"should ignore"}, []zapcore.Field{}, []observer.LoggedEntry{ignored}},
 		{[]interface{}{"foo", 42, "true", "bar"}, []zapcore.Field{Int("foo", 42), String("true", "bar")}, nil},
+		{[]interface{}{Int("foo", 42)}, []zapcore.Field{Int("foo", 42)}, nil},
+		{[]interface{}{Int("foo", 42), "should ignore"}, []zapcore.Field{Int("foo", 42)}, []observer.LoggedEntry{ignored}},
+		{
+			[]interface{}{"first", "field", Int("foo", 42), "baz", "quux", "should ignore"},
+			[]zapcore.Field{String("first", "field"), Int("foo", 42), String("baz", "quux")},
+			[]observer.LoggedEntry{ignored},
```

Comment:
I'm trying to sort out whether these cases test the development panic path, or the production path, but I can't figure it out just from these declarations:
- I think at this point we need to have a `desc string` field in each case
- bonus points if we switch to named-field struct literals, as this is getting a bit unweildy to read
- whichever path (dev v prod) is covered, cover the other one too, even if it means we need a whole other `func Test*` or a new dimension in our case structs